tests/acceptance/test_nidigital_measurement.py

```diff
@@ -0,0 +1,121 @@
+import pathlib
+from typing import Generator, Iterable, NamedTuple
+
+import pytest
+
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.measurement.v2.measurement_service_pb2 import (
+    MeasureRequest,
+)
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.measurement.v2.measurement_service_pb2_grpc import (
+    MeasurementServiceStub,
+)
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.pin_map_context_pb2 import (
+    PinMapContext,
+)
+from ni_measurementlink_service.measurement.service import MeasurementService
+from tests.assets.stubs.nidigital_measurement.types_pb2 import Configurations, Outputs
+from tests.utilities import nidigital_measurement
+from tests.utilities.pin_map_client import PinMapClient
+
+
+def test___single_session___measure___returns_measured_values(
+    pin_map_id: str,
+    stub_v2: MeasurementServiceStub,
+) -> None:
+    pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0])
+    configurations = Configurations(pin_names=["CS", "SCLK", "MOSI", "MISO"], multi_session=False)
+
+    outputs = _measure(stub_v2, pin_map_context, configurations)
+
+    assert outputs.passing_sites == [0]
+    assert outputs.failing_sites == []
+
+
+def test___single_session___measure___creates_single_session(
+    pin_map_id: str,
+    stub_v2: MeasurementServiceStub,
+) -> None:
+    pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0])
+    configurations = Configurations(pin_names=["CS", "SCLK", "MOSI", "MISO"], multi_session=False)
+
+    outputs = _measure(stub_v2, pin_map_context, configurations)
+
+    assert _get_output(outputs) == [
+        _MeasurementOutput(
+            "DigitalPattern1",
+            "DigitalPattern1",
+            "site0/CS, site0/SCLK, site0/MOSI, site0/MISO",
+            "site0/CS",
+        )
+    ]
+
+
+def test___multiple_sessions___measure___creates_multiple_sessions(
+    pin_map_id: str,
+    stub_v2: MeasurementServiceStub,
+) -> None:
+    pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0, 1])
+    configurations = Configurations(pin_names=["CS"], multi_session=True)
```

Comment:
This measurement bursts a pattern on the DUT's SPI bus pins, which uses all 4 SPI bus pins, not just CS. It doesn't make sense to reserve only CS. Please reserve all 4 pins.

This doesn't apply to the integration tests, which are testing lower-level functions and not running the measurement.

Reply:
Understood, updated the test to pass all the 4 pins.